### PR TITLE
sidekick: init at 0.6.6

### PR DESCRIPTION
**Pull request by nipeharefa**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/si/sidekick/package.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      --fish <($out/bin/sidekick completion fish) \
+      --zsh <($out/bin/sidekick completion zsh)
+  '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+
+  meta = {
+    description = "Bare metal to production ready in mins; your own fly server on your VPS.";
```
Comment (Kek5chen) [inline pkgs/by-name/si/sidekick/package.nix:45]: https://nixos.org/manual/nixpkgs/stable/#var-meta-description

**Comment by Kek5chen** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353505`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sidekick</li>
  </ul>
</details>

**Review by Kek5chen — CHANGES_REQUESTED** [review]
Description is not up to standards. Otherwise looks good.

File: pkgs/by-name/si/sidekick/package.nix (hunk 1 above)
Comment (nipeharefa) [reply pkgs/by-name/si/sidekick/package.nix:45]: on it

Comment (nipeharefa) [reply pkgs/by-name/si/sidekick/package.nix]: updated @Kek5chen, could you please check again

**Review by Kek5chen — APPROVED** [review]

File: pkgs/by-name/si/sidekick/package.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    installShellCompletion --cmd sidekick \
+      --bash <($out/bin/sidekick completion bash) \
+      --fish <($out/bin/sidekick completion fish) \
+      --zsh <($out/bin/sidekick completion zsh)
+  '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
```
Comment (Aleksanaa) [inline pkgs/by-name/si/sidekick/package.nix:41]: Also `doInstallCheck = true;`

File: pkgs/by-name/si/sidekick/package.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  ];
+
+  meta = {
+    description = "A command-line tool designed to simplify the process of deploying and managing applications on a VPS";
```
Comment (Aleksanaa) [inline pkgs/by-name/si/sidekick/package.nix:45]:
```suggestion
    description = "Command-line tool designed to simplify the process of deploying and managing applications on a VPS";
```

Comment (nipeharefa) [reply pkgs/by-name/si/sidekick/package.nix]: fixed @Aleksanaa

File: pkgs/by-name/si/sidekick/package.nix (hunk 2 above)
Comment (nipeharefa) [reply pkgs/by-name/si/sidekick/package.nix:43]: fixed @Aleksanaa

Comment (Kek5chen) [reply pkgs/by-name/si/sidekick/package.nix:43]: just wondering, but buildGoModule doesn't run install checks afaik, and there's no Makefile either, is this really necessary?

**Review by Aleksanaa — APPROVED** [review]

**Review by Kek5chen — APPROVED** [review]

**Comment by Aleksanaa** [comment]
Thanks, very useful

**Pull request merged**